das/local_file_storage_service.go

```diff
@@ -46,18 +44,20 @@ func NewLocalFileStorageService(dataDir string) (StorageService, error) {
 
 func (s *LocalFileStorageService) GetByHash(ctx context.Context, key []byte) ([]byte, error) {
 	log.Trace("das.LocalFileStorageService.GetByHash", "key", pretty.FirstFewBytes(key), "this", s)
-	s.mutex.RLock()
-	defer s.mutex.RUnlock()
 	pathname := s.dataDir + "/" + base32.StdEncoding.EncodeToString(key)
 	return os.ReadFile(pathname)
 }
 
 func (s *LocalFileStorageService) Put(ctx context.Context, data []byte, timeout uint64) error {
 	log.Trace("das.LocalFileStorageService.Store", "message", pretty.FirstFewBytes(data), "timeout", time.Unix(int64(timeout), 0), "this", s)
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
 	pathname := s.dataDir + "/" + base32.StdEncoding.EncodeToString(crypto.Keccak256(data))
-	return os.WriteFile(pathname, data, 0600)
+
+	err := os.WriteFile(pathname+"_temp", data, 0600)
```

Comment:
There is a race condition here if there are two simultaneous Puts of the same data. Both Puts will try to write to the same temp file. 

Two possible fixes are (1) use a Mutex to prevent simultaneous Puts, or (2) use `os.CreateTemp` to make a guaranteed-unique temp file name for each call.

Reply:
I had just thought about making writes atomic wrt reads, but you're right that we may have situations where we get multiple writes of the same data, so I made this change.